snuba/replacers/replacements_and_expiry.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime, timedelta
+from typing import Mapping, Sequence
+
+logger = logging.getLogger(__name__)
+
+
+from snuba.redis import RedisClientKey, get_redis_client
+from snuba.state import get_int_config
+
+redis_client = get_redis_client(RedisClientKey.REPLACEMENTS_STORE)
+config_auto_replacements_bypass_projects_hash = (
+    "snuba-config-auto-replacements-bypass-projects-hash"
+)
+
+REPLACEMENTS_EXPIRY_WINDOW_MINUTES_KEY = "replacements_expiry_window_minutes"
+
+
+def get_expiry_window_or_counter_window_size(key: str, default: int) -> int:
+    minutes = get_int_config(key=key, default=default)
+    return int(minutes) if minutes else default
```

Comment:
This function is not really abstracting much, it's also confusing because you use it within and outside this module for different reasons. It would make more sense to simply get this config where you actually need it rather than define a function which does not simplify anything

Reply:
Makes sense. The other way I can think of currently to get around the type checking is using `assert` statements. Let me know what you think

Reply:
you can just use `int(get_int_config(key, 20))` if you're passing an int as a default to `get_int_config` there's no way you're not getting back an int

Reply:
I agree that there's no way we get back `None` in practice, however directly casting it to `int` without either `assert` or the helper function approach results in this error: https://github.com/getsentry/snuba/actions/runs/9134800036/job/25121068644?pr=5883

Reply:
then you can use `typing.cast` or `# type: ignore`